docs/IAS-XSUAA-token-xchange.md

```diff
@@ -0,0 +1,66 @@
+# IAS to XSUAA Token exchange
+Token exchange supports principal propagation from an IAS-based app/service to an XSUAA-based service, logically staying in the same zone.
+This is a hybrid scenario of cross consumption between IAS and XSUAA.
+
+### Scenario
+The user is logged in to a multitenant application that has adopted IAS and zones.
+The application (or an internally used SCP service instance) calls a service in the context of the current user and zone, by sending the IAS ID token for the user. That means the IAS ID token consists of `user_uuid` and `zone_uuid` claim.
+The service instance uses XSUAA internally and thus needs to exchange the ID token from IAS to an access token from XSUAA. 
+In the XSUAA multitenancy concept, the zone is mapped to the single XSUAA tenant that belongs to the zone (=> XSUAA tenant ID = zone ID).
+
+### Under the hood
+![IAS -> XSUAA token xchange flow diagram](token-xchange.png)
+
+Token of incoming request is checked whether it is a Xsuaa token, if it is Xsuaa token then this token proceeds forward to the token validation step. In case it is valid token, successful response is sent back, otherwise an unauthorised response is sent. 
```

Comment:
authorised, authorisation => use american notion of it: authorized, authorization